lib/cache.js

```diff
@@ -26,22 +25,20 @@ Cache.prototype.add = function(
   this.set(key, val);
 };
 
-Cache.prototype.del = function(
-  // Delete cache element
-  key // string, key
-) {
+// Delete cache element
+//   key - string, key
+Cache.prototype.del = function(key) {
   if (this.has(key)) {
     const val = this.get(key);
     this.allocated -= dataSize(val);
   }
   this.delete(key);
 };
 
-Cache.prototype.clr = function(
-  // Clear cache elements that start with prefix
-  prefix, // string, to compare with beginning of the key
-  fn // function (optional), `(key, val)` to be called on each key
-) {
+// Clear cache elements that start with prefix
+//   prefix - string, to compare with beginning of the key
+//   fn - function (optional), `(key, val)` to be called on each key
```

Comment:
Strange ordering, shouldn't function signature be next to the function keyword as it's actually a type?

Reply:
I think we can just use 
`// fn - function(key, val), to be called on each key`
?

Reply:
Well, to be parsed correctly we have to have something like this
```
//   fn - function, function(key, val), to be called on each key
```
or
```
//   fn - function (optional), to be called on each key
//     key - string, key
//     val - value
```

Reply:
I'd prefer the second variant but add the signature in the first type
`fn - function(key, val) (optional)`
Also note that the comment to the function might be multiline.